be/src/olap/reader.cpp

```diff
@@ -624,17 +633,22 @@ OLAPStatus Reader::_init_keys_param(const ReaderParams& read_params) {
 
 void Reader::_init_conditions_param(const ReaderParams& read_params) {
     _conditions.set_tablet_schema(&_tablet->tablet_schema());
+    _all_conditions.set_tablet_schema(&_tablet->tablet_schema());
     for (const auto& condition : read_params.conditions) {
         ColumnPredicate* predicate = _parse_to_predicate(condition);
         if (predicate != nullptr) {
             if (_tablet->tablet_schema()
                         .column(_tablet->field_index(condition.column_name))
                         .aggregation() != FieldAggregationMethod::OLAP_FIELD_AGGREGATION_NONE) {
                 _value_col_predicates.push_back(predicate);
+                OLAPStatus status = _all_conditions.append_condition(condition);
+                DCHECK_EQ(OLAP_SUCCESS, status);
             } else {
                 _col_predicates.push_back(predicate);
                 OLAPStatus status = _conditions.append_condition(condition);
                 DCHECK_EQ(OLAP_SUCCESS, status);
+                status = _all_conditions.append_condition(condition);
```

Comment:
`_all_conditions.append_condition(condition);` can be moved outside the `if-else` block?